### ISSUE TITLE
Pasting files between VS Code windows is extremely slow

### ISSUE THREAD

**Issue by TomasHubelbauer** [Microsoft/vscode#72204]
I haven't seen an existing issue created for this yet. I have noticed that on my work laptop - which sucks to be clear, but is definitely still a type of machine people might realistically have and use, not that much of a relic - pasting files copied in another VS Code window is extremely slow. We're talking like two seconds a file here.

I think this might be because a new editor window is opened for each file pasted?

- VSCode Version: 1.33.1
- OS Version: Windows 10

Steps to Reproduce:

1. Copy files in window A
2. Paste files in window B
3. Observe files being pasted one by one every two seconds or so

Does this issue occur when all extensions are disabled?: Not sure

**Comment by isidorn** [comment]
I have refactored the paste command. And now we no longer open editors when pasting more than 1 file which I think makes sense.
@bpasero If we would want to make this really crisp and fast the file service would need to start supporting moving multiple resources at once.
Currently I can not call in parallel to move 12 resources, but have to sequantually do it one by one, which makes this operation feel slow.

@TomasHubelbauer anyways, try it out from tomorrow's vscode insiders and let us know what you think

**Comment by bpasero** [comment]
@isidorn I can look into providing a method to move multiple. This also applies to copy right? Anything else?

**Comment by bpasero** [comment]
@isidorn also, I assume you are using `ITextFileService.move()` right? Because that can take care of dirty files properly. Maybe check if that is the case and we can make that one bulk.

**Comment by bpasero** [comment]
> Currently I can not call in parallel to move 12 resources, but have to sequantually do it one by one, which makes this operation feel slow.

Actually, why is that? After all you should be able to do `Promise.join()` on these operations, no?

**Comment by isidorn** [comment]
@bpasero when I do `Promise.join` 80% of my calls are simply ignored. If you want I can file a new issue for this to you.
Currently I am using the `FileService`, not the `TextFileService`. I can move to ITextFileService if you provide me the bulk move.

Yes the paste implementation is shared between copy and cut. So this also applies to copy.

**Comment by bpasero** [comment]
@isidorn yes that would be a bug, can you try to come up with a test case that reproduces this issue? I have many tests for `move()` already [here](https://github.com/Microsoft/vscode/blob/407e266857d236bda5374d52708d9cdce356bea7/src/vs/workbench/services/files/test/node/diskFileService.test.ts#L439).

Yes, you should use `ITextFileService.move()` because that is the only one that takes care of dirty models (e.g. it ensures to preserve dirty state and contents). I am surprised you did not get a bug so far when the file is dirty?

**Comment by isidorn** [comment]
@bpceee probably everyone uses auto save like they should thus no bug for me :)
I have created this debt item https://github.com/Microsoft/vscode/issues/72543  which for now I assigned for May. To move to the ITextFileService and to write a test for move().
